environment/src/main/java/com/sequenceiq/environment/environment/poller/StackPollerProvider.java

```diff
@@ -0,0 +1,83 @@
+package com.sequenceiq.environment.environment.poller;
+
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.AttemptResults;
+import com.dyngr.core.AttemptState;
+import com.sequenceiq.environment.environment.service.stack.StackService;
+import com.sequenceiq.flow.core.FlowConstants;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.flowlog.FlowLogDBService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.ws.rs.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StackPollerProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackPollerProvider.class);
+
+    private final StackService stackService;
+
+    private final FlowLogDBService flowLogDBService;
+
+    public StackPollerProvider(
+        StackService stackService,
+        FlowLogDBService flowLogDBService) {
+        this.stackService = stackService;
+        this.flowLogDBService = flowLogDBService;
+    }
+
+    public AttemptMaker<Void> stackUpdateConfigPoller(List<String> stackNames, Long envId, String flowId) {
+        List<String> mutableNamesList = new ArrayList<>(stackNames);
+        return () -> {
+            Optional<FlowLog> flowLog = flowLogDBService.getLastFlowLog(flowId);
+            if (flowLog.isPresent() && flowLog.get().getCurrentState().equals(FlowConstants.CANCELLED_STATE)) {
```

Comment:
I tried that.  Couple of issues with it.  

Biggest issue is that it will not stop the 
flow if it is running on a different instance of the environment service. 
EnvironmentInMemoryStateStore just ends up as a local ConcurrentHashMap.

Other issue is that even locally you have no way to guarantee that the CANCELED value
is picked up by the sleeping polling instance before the the new Polling instance starts 
and resets it to POLLING since it is only referenced by environment id.
   
Both of these led to just checking the state of the flow and if it has been 
canceled or not.  This allows for cross instance control and explicit stopping 
of the prior flow instance.